### ISSUE TITLE
self-referential foreign key

### ISSUE THREAD

**Issue by Turbid** [digi604/django-smart-selects#128]
For code:
```
class Subdivision(models.Model):
    affiliate = ForeignKey(Group)
    parent = ChainedForeignKey("self", chained_field="affiliate", chained_model_field="affiliate",
                               show_all=False, auto_choose=True, null=True, blank=True, related_name="subdivisions")
    subdivision = CharField(verbose_name="subdivision", max_length=255)
```
I got this errror:
```
File "..\Python35\lib\site-packages\smart_selects\db_fields.py", line 154, in __init__
    self.to_app_name, self.to_model_name = to.split('.')
ValueError: not enough values to unpack (expected 2, got 1)
```
Because "self" have not '.' for split

**Comment by mpachas** [comment]
@Turbid I've been testing with that... does this work? It seems to work for me

```python
parent = ChainedForeignKey("yourAppName.Subdivision", chained_field="affiliate", chained_model_field="affiliate",
                               show_all=False, auto_choose=True, null=True, blank=True, related_name="subdivisions")
```

**Comment by blag** [comment]
@mpachas That should work, but I'm cooking up a proper fix. I should have it either in a PR or merged in by tomorrow.

**Comment by Turbid** [comment]
@blag yep, I'll do it tomorrow

**Comment by blag** [comment]
Awesome, thanks!

**Comment by mpachas** [comment]
i will check it too

**Comment by Turbid** [comment]
I try branch with class from topic, but got this error when I tried to add an object through the "add" button from django-admin:
`Reverse for 'chained_filter' with arguments '()' and keyword arguments '{'app': 'contacts', 'model': 'Subdivision', 'foreign_key_app_name': 'contacts', 'foreign_key_model_name': 'Subdivision', 'field': 'affiliate', 'value': '1', 'foreign_key_field_name': 'parent'}' not found. 0 pattern(s) tried: []`

**Comment by blag** [comment]
Did you follow step 2 of the [installation directions](https://github.com/digi604/django-smart-selects#installation)?

1. Add smart_selects to your INSTALLED_APPS
2. Bind the `smart_selects` urls into your project's `urls.py`. This is needed for the `ChainedSelects` and `ChainedManyToManySelects`. For example:

    ```python
    urlpatterns = patterns('',
        url(r'^admin/', include(admin.site.urls)),
        url(r'^chaining/', include('smart_selects.urls')),
    )
    ```

**Comment by Turbid** [comment]
Oh, this is my carelessness. Now all wonderful.
Thank you very much for the work done.

**Comment by blag** [comment]
No worries. So it works just fine then?

**Comment by Turbid** [comment]
Yes, everything is now as it should

**Comment by blag** [comment]
Thanks for testing - I just merged this and pushed it to PyPI as version 1.2.5.
